simulator/host_system.go

```diff
@@ -106,6 +135,241 @@ func (h *HostSystem) configure(spec types.HostConnectSpec, connected bool) {
 	id := newUUID(h.Name)
 	h.Summary.Hardware.Uuid = id
 	h.Hardware.SystemInfo.Uuid = id
+
+	var err error
+	h.sh, err = createSimulationHost(ctx, h)
+	if err != nil {
+		panic("failed to create simulation host and no path to return error: " + err.Error())
+	}
+}
+
+// configureContainerBacking sets up _this_ host for simulation using a container backing.
+// Args:
+//
+//		image - the container image with which to simulate the host
+//		mounts - array of mount info that should be translated into /vmfs/volumes/... mounts backed by container volumes
+//	 	networks - names of bridges to use for underlays. Will create a pNIC for each. The first will be treated as the management network.
+//
+// Restrictions adopted from createSimulationHost:
+// * no mock of VLAN connectivity
+// * only a single vmknic, used for "the management IP"
+// * pNIC connectivity does not directly impact VMs/vmks using it as uplink
+//
+// The pnics will be named using standard pattern, ie. vmnic0, vmnic1, ...
+// This will sanity check the NetConfig for "management" nicType to ensure that it maps through PortGroup->vSwitch->pNIC to vmnic0.
+func (h *HostSystem) configureContainerBacking(ctx *Context, image string, mounts []types.HostFileSystemMountInfo, networks ...string) error {
+	option := &types.OptionValue{
+		Key:   advOptContainerBackingImage,
+		Value: image,
+	}
+
+	advOpts := Map.Get(h.ConfigManager.AdvancedOption.Reference()).(*OptionManager)
```

Comment:
`ctx.Map`, same thing currently, but hopefully someday we'll get rid of the simulator.Map package var